### PR TITLE
Add support for shell completion

### PR DESCRIPTION
**Pull request by chevdor**
The Polkadot cli command now supports shell auto-completion. 
The auto-completion is now generated from the build. Once executed, this script show the following if you type `polkadot <tab><tab>`:
```
$ polkadot 
--base-path      --collator       --help           --light          --no-telemetry   --pruning        --telemetry-url  --ws-port        -h               build-spec       import-blocks    
--bootnodes      --dev            --key            --log            --node-key       --rpc-port       --validator      -V               -l               export-blocks    
--chain          --execution      --keystore-path  --name           --port           --telemetry      --version        -d               -t               help     
```

The completion script is generated from `cli.yml` so it will keep in sync as the command and flags evolve.

See it in action below:
[asciicast](https://asciinema.org/a/192080?autoplay=1)

Assuming you built a release version using `cargo build --release` and use `bash` run the followig:
```
source target/release/completion-scripts/polkadot.bash
```

You can find completion scripts for:
- bash
- fish
- zsh
- elvish
- powershell

### CODE REVIEW

**Comment by tomusdrw** [comment]
Maybe better to add a CLI option to generate that in runtime?
```
polkadot cli-completion > polkadot.bash
```
Makes different version management easier, since you just obtain the binary and generate the completions for yourself locally.

**Comment by chevdor** [comment]
Hi @tomusdrw, thanks for your feedback.

I was considering it but went for a build only option in order to not overload the cli with code that is used only once at most and keep the polkadot cli as slim as possible but this is something that can be changed of course.

It may be even worth having this as feature.

If we decide to augment the cli, we will also need to pass and validate the shell:
```
polkadot cli-completion bash > polkadot.bash
polkadot cli-completion powershell > polkadot.ps1
...
```

File: README.md [hunk 1]
```diff
@@ -101,3 +101,39 @@ You can start a development chain with:
 ```
 cargo run -- --chain=dev --validator --key Alice
 ```
+
+## Shell completion
+
+The Polkadot cli command supports shell auto-completion. For this to work, you will need to run the completion script matching you build and system.
+
+Assuming you built a release version using `cargo build --release` and use `bash` run the followig:
```
Comment (rphmeier) [inline README.md]: typo: `followig`

File: polkadot/cli/build.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    };
+    let path = Path::new(&outdir)
+    	.parent().unwrap()
+    	.parent().unwrap()
+    	.parent().unwrap()
+    	.join("completion-scripts");
+
+    fs::create_dir(&path).ok();
+
+    // println!("{:?}", path);
```
Comment (rphmeier) [inline polkadot/cli/build.rs]: remaining commented println

File: README.md (hunk 1 above)
Comment (chevdor) [reply README.md]: Thanks, fixed.

**Pull request merged**